go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -932,7 +932,23 @@ func (tsv *TabletServer) ExecuteBatch(ctx context.Context, target *querypb.Targe
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "cannot start a new transaction in the scope of an existing one")
 	}
 
+	if tsv.enableHotRowProtection && asTransaction && len(queries) == 1 {
+		// Serialize transactions which target the same hot row range.
+		// NOTE: We put this intentionally at this place *before* tsv.startRequest()
+		// gets called below. Otherwise, startRequest() from below would cross with
+		// the startRequest() call executed by tsv.beginWaitForSameRangeTransactions().
+		txDone, err := tsv.beginWaitForSameRangeTransactions(ctx, target, options, queries[0].Sql, queries[0].BindVariables)
+		if err != nil {
+			return nil, err
+		}
+		if txDone != nil {
+			defer txDone()
+		}
+	}
+
 	allowOnShutdown := (transactionID != 0)
+	// TODO(alainjobart): Convert startRequest/endRequest pattern to use wrapper
```

Comment:
Hey don't add TODO for me :)

Reply:
That's officially allowed now :) You're supposed to assign the TODO to the most suitable person who can fix this :)

Reply:
Well send it to @sougou then, I'm not the best person! ;)

Reply:
Done.